bin/markcop

```diff
@@ -25,6 +26,80 @@ errors=''
 IFS='
 '
 
+# Check for trailing whitespace
+printf '\n\nChecking for trailing whitespace...\n'
+
+for line in $(git ls-files | grep '\.md$' | xargs grep -noP '^.+\s$' | sed -e 's/:/\t/' -e 's/:/\t/'); do
+  file_name=$(echo $line | cut -f1)
+  line_num=$(echo $line | cut -f2)
+  line_contents=$(echo $line | cut -f3)
+  HAS_ERROR=true
+  trailing_whitespace=true
+  printf "${RED}x${NC}"
+  errors="${errors}\n${line_num}:${file_name} '$line_contents' has trailing whitespace!"
+done
+
+if [[ ! $trailing_whitespace = true ]]; then
+  printf "${GREEN}.${NC}\nNo trailing whitespace!"
+fi
+
+# Check that headings have proper formatting
+printf '\n\nChecking for incorrect headings...\n'
+
+for line in $(git ls-files | grep '\.md$' | xargs grep -noP '^\#[^\{]*$' | sed -e 's/:/\t/' -e 's/:/\t/'); do
+  bad_heading=false
+  file_name=$(echo $line | cut -f1)
+  line_num=$(echo $line | cut -f2)
+  line_contents=$(echo $line | cut -f3)
+  if [[ $line_contents =~ ^\#{1,6}[^\ \#].*$ ]]; then
+    bad_heading=true
+    HAS_BAD_HEADING=true
+    printf "${RED}x${NC}"
+    errors="${errors}\n${line_num}:${file_name} '${line_contents}' doesn't have a space between the # and the text of the header!"
+  fi
+  if [[ $line_contents =~ ^\#{7,}.+$ ]]; then
```

Comment:
`^\#{7,}.*$` is probably more appropriate here.

`#######` would pass the `.+` version.